src/cargo/ops/registry.rs

```diff
@@ -55,6 +57,39 @@ pub fn publish(manifest_path: &Path,
     Ok(())
 }
 
+fn check_git_cleanliness(current_path: &Path) -> CargoResult<()>{
+    match git2::Repository::open(current_path){
```

Comment:
Correct. I have a test project to and if i run the program from a sub directory it does not detect git. I am going to look into using open_ext http://alexcrichton.com/git2-rs/git2/struct.Repository.html#method.open_ext  it sounds like what i want.